.github/workflows/scheduled.yml

```
@@ -218,6 +218,40 @@ jobs:
       - name: Deploy to GitHub Pages
         id: deployment
         uses: actions/deploy-pages@v1
+  report-status:
+    name: Report deploy status
+    needs: [generate-cask, generate-core, generate-analytics, generate-samples, build, deploy]
```

```suggestion
    needs: deploy
```

Think this should be enough?

---

No, we need all of the recursive dependencies because otherwise the `needs` context will only contain the `deploy` job.

Later, we have a condition of:

```yml
if: ${{ contains(needs.*.result, 'failure') }}
```

In the case where `generate-cask` fails, all subsequent jobs (importantly `build` and `deploy`) will be marked as `skipped` rather than `failed`. We could in theory check for `needs.deploy.result == 'skipped'` as well as `failed`, but I worry that there are more edge cases where skipped shows up but we don't want it to be treated as a failure. For example, maybe a new run was triggered and the concurrency rules canceled the old run. In that case, it may not be a true failure so we probably don't want to do anything.

If we just have `needs: deploy`, we can't check these other dependency jobs for their status (at least as I understand it).

---

Gotcha, works for me!